File: problems/trademarket/solutions/bad.js
const getMaxProfit = input => {
    const prices = input.split(' ').map(price => parseInt(price));

    let maxProfit = prices[1] - prices[0];
    let currentMin = prices[0] + 1;

    for (let i = 0; i < prices.length; i++) {
        if (prices[i] < currentMin) {
            currentMin = prices[i];
            for (let j = i + 1; j < prices.length; j++) {
                maxProfit = Math.max(maxProfit, prices[j] - currentMin);
            }
        }
    }

    return maxProfit;
}



const chunks = [];
process.stdin.on('data', d => chunks.push(d));

process.stdin.on('end', () => {
    const data = chunks.join('').trim();
    console.log(getMaxProfit(data));
});

File: problems/trademarket/solutions/bad.py
def get_max_profit(prices):
    max_profit = prices[1] - prices[0]
    current_min = prices[0] + 1

    for i, price in enumerate(prices):
        if price < current_min:
            current_min = price
            for price_2 in prices[i + 1:]:
                max_profit = max(max_profit, price_2 - current_min)

    return max_profit

prices = [int(x) for x in input().split()]

print(get_max_profit(prices))